include/revng/FunctionIsolation/InvokeIsolatedFunctions.h

```diff
@@ -11,18 +11,19 @@
 #include "revng/BasicAnalyses/GeneratedCodeBasicInfo.h"
 #include "revng/Model/LoadModelPass.h"
 
-class InvokeIsolatedFunctionsPass : public llvm::ModulePass {
+class InvokeIsolatedFunctionsPass : public pipeline::ModulePass {
 public:
   static char ID;
 
 public:
-  InvokeIsolatedFunctionsPass() : ModulePass(ID) {}
+  InvokeIsolatedFunctionsPass() : pipeline::ModulePass(ID) {}
 
-  bool runOnModule(llvm::Module &M) override;
+  bool run(llvm::Module &M, const pipeline::TargetsList &Targets) override;
 
   void getAnalysisUsage(llvm::AnalysisUsage &AU) const override {
     AU.addRequired<GeneratedCodeBasicInfoWrapperPass>();
     AU.addRequired<LoadModelWrapperPass>();
+    AU.addRequired<pipeline::LoadExecutionContextPass>();
```

Comment:
Why do we still need `LoadModelWrapperPass`?
We can get the model from the `ExecutionContext`, can't we?

Reply:
yes, but we need to schedule LOadModelWrapperPAss anyway so people can do both. Do we wish to have a policy about this? in principle we could even entirely drop LoadModelWrapperPass

Reply:
i looked into this, but the change was sprawling and it still requires to make sure that revng opt is never used. I recommend to do it after after that issue is fixed